# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/cookiebanners/test/browser/browser_cookiebanner_gdpr_telemetry.js
# Commit: 9f420ea15109
# Full Hash: 9f420ea1510986e266160d52d176ed7227ef5e40
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2024-03-30 09:10:19
# Regressor Bug: 1874741
# File Overlap Count: 1
# Description:
#   Bug 1874741 - Part 5: Implement the GDPR event telemetry for private browsing windows. r=pbz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D205900
# ==============================================================================

diff -r d977620d7182 -r 9f420ea15109 toolkit/components/cookiebanners/test/browser/browser_cookiebanner_gdpr_telemetry.js
--- a/toolkit/components/cookiebanners/test/browser/browser_cookiebanner_gdpr_telemetry.js	Fri Mar 29 23:15:12 2024 +0000
+++ b/toolkit/components/cookiebanners/test/browser/browser_cookiebanner_gdpr_telemetry.js	Fri Mar 29 23:15:13 2024 +0000
@@ -139,3 +139,51 @@
     "No event telemetry is recorded."
   );
 });
+
+add_task(async function test_google_gdpr_telemetry_pbm() {
+  // Clear telemetry before starting telemetry test.
+  Services.fog.testResetFOG();
+
+  for (let test of TEST_CASES) {
+    // Add a private Google SOCS cookie to trigger recording telemetry.
+    SiteDataTestUtils.addToCookies({
+      name: "SOCS",
+      value: test.value,
+      host: test.host,
+      path: "/",
+      originAttributes: { privateBrowsingId: 1 },
+    });
+
+    // Ensure we don't record google GDPR choice cookie telemetry.
+    is(
+      Glean.cookieBanners.googleGdprChoiceCookie[test.host].testGetValue(),
+      null,
+      "No Google GDPR telemetry is recorded."
+    );
+
+    // Verify the the event telemetry for PBM is recorded properly.
+    let events =
+      Glean.cookieBanners.googleGdprChoiceCookieEventPbm.testGetValue();
+    let event = events[events.length - 1];
+
+    is(
+      event.extra.choice,
+      test.expected,
+      "The Google GDPR event telemetry records the choice properly."
+    );
+
+    is(event.extra.search_domain, undefined, "No search domain is recorded.");
+    is(event.extra.region, undefined, "No region is recorded.");
+  }
+
+  is(
+    Glean.cookieBanners.googleGdprChoiceCookieEventPbm.testGetValue().length,
+    TEST_CASES.length,
+    "The number of events is expected."
+  );
+
+  // We need to notify the "last-pb-context-exited" tp explicitly remove private
+  // cookies. Otherwise, the remaining private cookies could affect following
+  // tests. The SiteDataTestUtils.clear() doesn't clear for private cookies.
+  Services.obs.notifyObservers(null, "last-pb-context-exited");
+});
